### PR TITLE
Modify start/stop to use LifecycleException instead of ListenerException

### PR DESCRIPTION
**Pull request by evandongen**

### CODE REVIEW

**Review by tnleeuw — COMMENTED** [review]
I'm not yet sure that I find the undeclared runtime exception `LifecycleException` better than the declared `ListenerException`.
We still have to catch that exception in several places, but we don't know from the method signature that that is so.

I do see that there have already been more changes towards this direction, but I'm just unsure it's the right direction.

File: core/src/main/java/org/frankframework/jms/JmsListenerBase.java [hunk 1]
```diff
@@ -103,17 +104,17 @@ public void configure() throws ConfigurationException {
 	}
 
 	@Override
-	public void start() throws ListenerException {
+	public void start() {
 		try {
 			super.start();
-		} catch (Exception e) {
-			throw new ListenerException("error opening listener [" + getName() + "]", e);
+		} catch (LifecycleException e) {
```
[the remark this answers is not in the document]
Comment (evandongen) [reply core/src/main/java/org/frankframework/jms/JmsListenerBase.java:110]: That is already catched in the super `org.frankframework.jms.JMSFacade#start`

**Comment by nielsm5** [comment]
> I'm not yet sure that I find the undeclared runtime exception `LifecycleException` better than the declared `ListenerException`. We still have to catch that exception in several places, but we don't know from the method signature that that is so.
> 
> I do see that there have already been more changes towards this direction, but I'm just unsure it's the right direction.

With senders, the `MessageSendingPipe` is responsible for starting and stopping all senders. In short, that should be the only(!) place which has to catch these exceptions (for senders). The same can happen for `Listeners` in the Receiver (and `MessageSendingPipe`).

File: core/src/main/java/org/frankframework/jms/JmsListenerBase.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (nielsm5) [reply core/src/main/java/org/frankframework/jms/JmsListenerBase.java:110]: Het is toch helemaal niet logisch een LifecycleException af te vangen en hem opnieuw te gooien?

File: cmis/src/main/java/org/frankframework/extensions/cmis/server/CmisEventDispatcher.java [hunk 2]
```diff
@@ -54,10 +55,12 @@ public static synchronized CmisEventDispatcher getInstance() {
...
 	}
 
-	public void registerEventListener(CmisEventListener listener) throws ListenerException {
+	public void registerEventListener(CmisEventListener listener)  {
 		CmisEvent event = listener.getEvent();
-		if(event == null)
-			throw new ListenerException("cannot register EventListener without event to listen on");
+
+		if (event == null) {
+			throw new LifecycleException("cannot register EventListener without event to listen on");
```
[the remark this answers is not in the document]
Comment (evandongen) [reply cmis/src/main/java/org/frankframework/extensions/cmis/server/CmisEventDispatcher.java:62]: Hmm, maar dan moet je die (checked) exception weer afvangen in `org.frankframework.extensions.cmis.CmisEventListener#start`. Aan de andere kant, in `CmisEventListener#configure` is dezelfde check al gedaan icm een `ConfigurationException` .. misschien kunnen we het throwen hier wel achterwege laten.

**Review by nielsm5 — APPROVED** [review]
Volgens mij is het goed zo?

**Review by tnleeuw — APPROVED** [review]

**Pull request merged**